### PR TITLE
DOCSP-34766 Improve VSCode Connection Experience

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

- General cleanup efforts and stylist updates to the connect page.
- Standardized H1's and left hand TOC naming to `Connection` based tasks.
- Rearranged tasks on page to be in a more use case friendly order.
- Updated connect to deployment task to mention the new connection form.
- Added a task for `Edit Connection`, newly added in 1.5 to the connect page. 
- Added a task for `Copy Connection String` previously undocumented to the connect page.
- Added `OIDC` and `AWS IAM` authentication options to the authentication section of the connection task.
- Added Advanced Connection Options `Replica Set Name`, `Default Authentication Database`, and `URI Options`.

> :red_circle: **I made an epic[ DOCSP-36125](https://jira.mongodb.org/browse/DOCSP-36215) to capture the refactor needs for this page / repo**:
 Ideally, each of these tasks would be separate. Currently, we have several unrelated tasks such as edit / rename / copy on the connection page. These should be single tasks page under a "Manage your Connections" drawer.

## STAGING

https://preview-mongodbianfmongodb.gatsbyjs.io/mongodb-vscode/DOCSP-34766-b/connect/#connect-to-a-deployment

https://preview-mongodbianfmongodb.gatsbyjs.io/mongodb-vscode/DOCSP-34766-b/connect/#edit-a-connection

https://preview-mongodbianfmongodb.gatsbyjs.io/mongodb-vscode/DOCSP-34766-b/connect/#copy-a-connection

## JIRA

https://jira.mongodb.org/browse/DOCSP-34766

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=65c3f2a66b2f17c95aebcde5

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Is this free of spelling errors?
- [x] Is this free of grammatical errors?
- [x] Is this free of staging / rendering issues?
- [x] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml [hunk 1]
```diff
@@ -180,52 +196,57 @@ content: |
...
+                 :manual:`SCRAM-SHA-256 </core/security-scram/>` as its
                  authentication mechanism.
                  
                  Provide the :guilabel:`Username`, :guilabel:`Password`,
                  and :manual:`Authentication Database
                  </core/security-users/#user-authentication-database>`
                  to authenticate the user.
 
-                 .. note::
+             - id: OIDC
```
Comment (ianf-mongodb) [inline source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml:214]: > :blue_circle: **I made an epic: I rearrange the options here to match how they look in the VSCode GUI:

Screenshot 2024-02-02 at 10 31 20 AM

File: source/connect.txt [hunk 2]
```diff
@@ -152,40 +239,42 @@ To rename a connection to a deployment in |vsce|:
...
 
 .. tabs::
 
-   .. tab:: Disconnect with the Command Palette
-      :tabid: command-palette
-
-      .. include:: /includes/steps/disconnect-command-palette.rst
-
-   .. tab:: Disconnect from the MongoDB View
+   .. tab:: With the MongoDB View
```
Comment (ianf-mongodb) [inline source/connect.txt:253]: *Note* - Rearrange all tasks to show the `MongoDB View` as the first tab since it is the easier of the two options.

File: source/connect.txt [hunk 3]
```diff
@@ -141,6 +150,84 @@ the shell connected to your selected deployment.
...
+               You can also click the :guilabel:`Edit Connection String`
+               toggle in the top-right of the form to edit the connection
+               string URI directly.
+
+            Once you are done editing the connection options click
+            :guilabel:`Save & Connect`.
+
+.. _vsce-copy-connection-string-task:
+
+Copy a Connection
```
Comment (ianf-mongodb) [inline source/connect.txt:200]: *Note* - Copy a Connection was previously undocumented, added to this PR.

File: source/connect.txt [hunk 4]
```diff
@@ -141,6 +150,84 @@ the shell connected to your selected deployment.
 
    :mongosh:`Perform CRUD Operations in the MongoDB Shell </crud>`
 
+
+.. _vsce-edit-connection-task:
+
+Edit a Connection
```
Comment (ianf-mongodb) [inline source/connect.txt:156]: *Note* - Edit a Connection is a new task introduced in 1.5.

File: source/connect.txt [hunk 5]
```diff
@@ -58,42 +58,51 @@ Considerations
...
 
-|vsce| provides two methods to create a connection to a deployment. You 
-can: 
+.. list-table::
+   :header-rows: 1
+   :widths: 20 40
 
-- Provide a deployment connection string, or 
-- Fill in a deployment information in specific fields.
+   * - Connection Method
```
Comment (ianf-mongodb) [inline source/connect.txt:70]: *Note* - Slight cleanup for readability here.

File: source/connect.txt [hunk 6]
```diff
@@ -58,42 +58,51 @@ Considerations
 
 .. _vsce-connect-task:
 
-Create a Connection to a Deployment
------------------------------------
+Create a Connection
```
Comment (jeff-allen-mongo) [inline source/connect.txt:61]: Just curious, are there plans to separate these sections into individual task pages at some point?

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml [hunk 7]
```diff
@@ -4,14 +4,34 @@ source:
...
 ---
-title: Open the |vsce| *Overview* page from the Command Palette or from 
-       the :guilabel:`MongoDB` view in the :guilabel:`Activity Bar`.
+title: Open the |vsce| Overview page
 level: 4
 ref: create-connection
 content: |
 
+  You can use the the MongoDB view or the Command Palette to open the 
+  overview page. Use the tabs below to see each procedure:
```
Comment (jeff-allen-mongo) [inline source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml:13]: [Minor]

> Use the tabs below to see each procedure:

Generally we don't want to use "below" to locate information within a topic.

https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#b

Maybe we can remove this sentence entirely, since it should be intuitive that users should select the appropriate tab.

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml [hunk 8]
```diff
@@ -180,52 +196,57 @@ content: |
...
-                 :guilabel:`Password` to authenticate
-                 the user.
+                    * - Consider Target Endpoint Trusted
+                      - Optional. Allows connecting to a target endpoint that is not in the 
+                        list of endpoints that are considered trusted by default. Only use 
+                        this option when connecting to servers that you trust.
+
+                    * - Enable Device Authentication Flow
+                      - Optional. When the :ref:`Show Device Auth Flow Checkbox <compass-settings-reference>` 
+                        setting is enabled, |compass-short| can provide you with a URL and code
```
Comment (jeff-allen-mongo) [inline source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml:245]: This replacement isn't working in staging, and I don't think we should use |compass-short|. It's unfortunate that our builds don't report this as an error.

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml [hunk 9]
```diff
@@ -246,6 +267,50 @@ content: |
...
+                    :widths: 40 20 40
+
+                    * - Field
+                      - Necessity
+                      - Description
+
+                    * - AWS Access Key Id
+                      - Required
+                      - An access key ID is a unique identifier 
+                        associated with your AWS account.
```
Comment (jeff-allen-mongo) [inline source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml:300]: Can we shorten this to:

"Unique identifier associated with your AWS account." ?

Suggest making similar edits to the other AWS field descriptions.

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml [hunk 10]
```diff
@@ -246,6 +267,50 @@ content: |
...
+
+                    * - AWS Secret Access Key
+                      - Optional
+                      - The secret access key is a long-term 
+                        credential associated with the access key ID.
+
+                    * - AWS Access Key Id
```
Comment (jeff-allen-mongo) [inline source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml:307]: We have AWS Access Key Id twice. Is that correct?

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml [hunk 11]
```diff
@@ -467,8 +532,34 @@ content: |
   If omitted, defaults to :guilabel:`Primary`. To learn more about read 
   preferences, see :manual:`Read Preference </core/read-preference/>`. 
 
+  You can select additional advanced options :guilabel:`Replica Set Name`, 
+  :guilabel:`Default Authentication Database` and :guilabel:`URI Options` 
+  which specify additional behavior of your MongoDB connection:
```
Comment (jeff-allen-mongo) [inline source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml:537]: I don't think we need to list out the options in the intro to this table. Maybe we can shorten this whole thing to:

"You can specify additional MongoDB connection behavior with the following options: "

**Review by jeff-allen-mongo — CHANGES_REQUESTED** [review]
@ianf-mongodb thanks for these updates. I left a some comments and questions for you to consider.

Jeff

File: source/connect.txt (hunk 6 above)
Comment (ianf-mongodb) [reply source/connect.txt:61]: Yes definitely, I put a note on the PR and already created an epic for the effort, I hope to revisit this when some of the Q1 projects are knocked out: 

🔴 I made an epic[ DOCSP-36125](https://jira.mongodb.org/browse/DOCSP-36215) to capture the refactor needs for this page / repo:
Ideally, each of these tasks would be separate. Currently, we have several unrelated tasks such as edit / rename / copy on the connection page. These should be single tasks page under a "Manage your Connections" drawer.

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml (hunk 7 above)
Comment (ianf-mongodb) [reply source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml]: Ah thanks for the reference, yes I agree this can be removed entirely.

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml (hunk 8 above)
Comment (ianf-mongodb) [reply source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml]: Doh, yes good catch, this was C+P from Compass due to the fact that now both apps share the same connection form. And yes, reliable logs would be super helpful to spot these sort of things.

File: source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml (hunk 10 above)
Comment (ianf-mongodb) [reply source/includes/steps-starting-vsce-connect-with-advanced-connection-settings.yaml]: Yes, this is suppose to be session token, good catch.

File: source/includes/steps-starting-vsce-connect-with-connection-string.yaml [hunk 12]
```diff
@@ -35,6 +35,31 @@ content: |
 
   .. tabs:: 
 
+     .. tab:: Open from the Activity Bar
+        :tabid: mdbview 
+
+         a. Select the |vscode| icon from the left side menu bar
```
[the remark this answers is not in the document]
Comment (ianf-mongodb) [reply source/includes/steps-starting-vsce-connect-with-connection-string.yaml]: Updated. Funnily enough I realized we dont have a MongoDB icon available in our [LG Icon Assets](https://www.mongodb.com/docs/meta/reference/icon/#rendered-icon-table) 😆 .

**Review by ianf-mongodb — COMMENTED** [review]
Thanks for the review @jeff-allen-mongo, I incorporated your feedback and suggestions. I am seeing one additional build error now:

```
ERROR(includes/steps-starting-vsce-connect-with-connection-string.yaml:6ish): Unexpected indentation
    Did you mean: .. blockquote::
```

But its not obvious for my edits where this is coming from, if you happen to notice please LMK I will be checking.

File: source/connect.txt (hunk 6 above)
Comment (jeff-allen-mongo) [reply source/connect.txt:61]: Oh sorry, missed that note in the PR. Thanks!

File: source/connect.txt [hunk 13]
```diff
@@ -58,42 +58,51 @@ Considerations
...
-:ref:`setting <vsce-settings>`:
+   * - :guilabel:`Advanced Connection Settings`
+     - This method lets you build a connection string with a form.
+       Use this option if you need to customize the connection 
+       string and want to see available connection options.
 
-.. include:: includes/connection-location-table.rst
+.. note::
+
+   To specify where the connections defined in |vsce| are saved, use
```
Comment (jeff-allen-mongo) [inline source/connect.txt:86]:
```suggestion
   To specify where |vsce| connections are saved, use 
```

File: source/connect.txt [hunk 14]
```diff
@@ -152,40 +239,42 @@ To rename a connection to a deployment in |vsce|:
 
 .. _vsce-disconnect-task:
 
-Disconnect from a Deployment
+Disconnect from a Connection
```
Comment (jeff-allen-mongo) [inline source/connect.txt:242]: This edit seems a little off to me. Why is are we replacing "Connection" with "Deployment" here?

File: source/connect.txt (hunk 2 above)
Comment (jeff-allen-mongo) [inline source/connect.txt:253]: Suggest removing "With the" from these tab headings.

```suggestion
   .. tab:: MongoDB View
```

File: source/connect.txt [hunk 15]
```diff
@@ -152,40 +239,42 @@ To rename a connection to a deployment in |vsce|:
...
-
-      .. include:: /includes/steps/disconnect-command-palette.rst
-
-   .. tab:: Disconnect from the MongoDB View
+   .. tab:: With the MongoDB View
       :tabid: mdb-view
 
       .. include:: /includes/steps/disconnect-mongodb-view.rst
 
+   .. tab:: With the Command Palette
```
Comment (jeff-allen-mongo) [inline source/connect.txt:258]:
```suggestion
   .. tab:: Command Palette
```

**Review by jeff-allen-mongo — APPROVED** [review]
LGTM % nits. Thanks @ianf-mongodb

File: source/connect.txt (hunk 14 above)
Comment (ianf-mongodb) [reply source/connect.txt]: This was to match the verbiage of the other subtasks, but I can agree it sounds a little odd so I will roll it back. @jeff-allen-mongo

**Comment by ianf-mongodb** [comment]
Hi @Anemy this is ready for review.
